staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fq/fq.go

```diff
@@ -0,0 +1,468 @@
+package fq
+
+import (
+	"math"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/clock"
+)
+
+// FQScheduler is a fair queuing implementation designed for the kube-apiserver.
+// FQ is designed for
+// 1) dispatching requests to be served rather than packets to be transmitted
+// 2) serving multiple requests at once
+// 3) accounting for unknown and varying service times
+// implementation of:
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md
+type FQScheduler struct {
```

Comment:
it's supposed to implement the `FairQueuingSystem` interface, isn't it?

Reply:
updated